docs/man.jpconj/man3/Style.3
.TH "Style" 3 "Tue Aug 29 2017" "Version 2.0.0" "JapKatsuyou.JpConj" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Style \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <style\&.h>\fP
.PP
Inherits \fBQObject\fP\&.
.SS "Signals"

.in +1c
.ti -1c
.RI "void \fBstyleChanged\fP (QString styleID)"
.br
.ti -1c
.RI "void \fBfontChanged\fP (QString font, QString jpfont, int size, int jpsize)"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBStyle\fP ()"
.br
.ti -1c
.RI "void \fBemitSignal\fP (QString styleID)"
.br
.ti -1c
.RI "void \fBsignalFontChanged\fP (QString font, QString jpfont, int size, int jpsize)"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static void \fBloadStyles\fP ()"
.br
.ti -1c
.RI "static QHash< QString, \fBStyleinfo\fP * > \fBgetStyleInfo\fP ()"
.br
.RI "\fI\fBStyle::getStyleInfo\fP Get ID's and names of available style sheets\&. \fP"
.ti -1c
.RI "static void \fBsetStyle\fP ()"
.br
.ti -1c
.RI "static void \fBsetFont\fP ()"
.br
.ti -1c
.RI "static QString \fBgetConfigStyle\fP ()"
.br
.ti -1c
.RI "static QString \fBgetCurrentStyle\fP ()"
.br
.ti -1c
.RI "static void \fBsetConfigStyle\fP (QString styleID)"
.br
.ti -1c
.RI "static void \fBaddReceiver\fP (const \fBQObject\fP *receiver, const char *receiverSlot)"
.br
.ti -1c
.RI "static void \fBaddFontReceiver\fP (const \fBQObject\fP *receiver, const char *receiverSlot)"
.br
.ti -1c
.RI "static QString \fBgetConfigFont\fP (bool jap)"
.br
.ti -1c
.RI "static QString \fBgetCurrentFont\fP (bool jap)"
.br
.ti -1c
.RI "static void \fBsetConfigFont\fP (QString font, bool jap)"
.br
.ti -1c
.RI "static int \fBgetConfigFontSize\fP (bool jap)"
.br
.ti -1c
.RI "static int \fBgetCurrentFontSize\fP (bool jap)"
.br
.ti -1c
.RI "static void \fBsetConfigFontSize\fP (int fontSize, bool jap)"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class is used to manage the tables styles (CSS) 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Style::Style ()"

.SH "Member Function Documentation"
.PP 
.SS "void Style::addFontReceiver (const \fBQObject\fP * receiver, const char * receiverSlot)\fC [static]\fP"

.SS "void Style::addReceiver (const \fBQObject\fP * receiver, const char * receiverSlot)\fC [static]\fP"

.SS "void Style::emitSignal (QString styleID)"

.SS "void Style::fontChanged (QString font, QString jpfont, int size, int jpsize)\fC [signal]\fP"

.SS "QString Style::getConfigFont (bool jap)\fC [static]\fP"

.SS "int Style::getConfigFontSize (bool jap)\fC [static]\fP"

.SS "QString Style::getConfigStyle ()\fC [static]\fP"

.SS "QString Style::getCurrentFont (bool jap)\fC [static]\fP"

.SS "int Style::getCurrentFontSize (bool jap)\fC [static]\fP"

.SS "QString Style::getCurrentStyle ()\fC [static]\fP"

.SS "QHash< QString, \fBStyleinfo\fP * > Style::getStyleInfo ()\fC [static]\fP"

.PP
\fBStyle::getStyleInfo\fP Get ID's and names of available style sheets\&. 
.PP
\fBReturns:\fP
.RS 4
A QHash of style-ID as key, and style-name as value\&. 
.br
Example: <DzStyle, DzCoding default style>, etc\&. 
.RE
.PP

.SS "void Style::loadStyles ()\fC [static]\fP"

.SS "void Style::setConfigFont (QString font, bool jap)\fC [static]\fP"

.SS "void Style::setConfigFontSize (int fontSize, bool jap)\fC [static]\fP"

.SS "void Style::setConfigStyle (QString styleID)\fC [static]\fP"

.SS "void Style::setFont ()\fC [static]\fP"

.SS "void Style::setStyle ()\fC [static]\fP"

.SS "void Style::signalFontChanged (QString font, QString jpfont, int size, int jpsize)"

.SS "void Style::styleChanged (QString styleID)\fC [signal]\fP"


.SH "Author"
.PP 
Generated automatically by Doxygen for JapKatsuyou\&.JpConj from the source code\&.


CHANGELOG.md

# json-VCard Changelog

This is the changelog syntax:
* NEW: For any new module or file (creation)
* ADD: For any functions added inside a module (addition)
* IMPROVE: For any improvements to existing materials.
This affects the behavior, but doesn't break any precedent
* MODIFY: Modification of function names, files names and directories.
This will break the projects developed on precedent versions.
* FIX: Fixing bugs
* DEL: Delete or deprecate some functions

## Version 2.0.0 (2016-07-03)

* ADD: Control the font (size and type) for both default language and Japanese
* NEW: Choose the icons set
* NEW: Show examples in default language
* ADD: Show Furigana and Romaji pronunciation of the verb

## Version 1.0.1 (2014-01-01)

* I don't know what changed :)

## Version 0.2.0 (2013-09-04)

* NEW: first release


jpconj/UiHelper/style.cpp
/*
    This file is part of JapKatsuyou project; an application that provide
    Japanese verb conjugation

    Copyright (C) 2013-2016  AnqaFalak group
    Copyright (C) 2013  Abdelkrime Aries <kariminfo0@gmail.com>

    Authors:
            Abdelkrime Aries <kariminfo0@gmail.com>

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */

#include "style.h"

/*!
 * \class Style
 * This class is used to manage the tables styles (CSS)
 */

QHash<QString, Styleinfo*> Style::styles;
QString Style::currentstyle = "";

QString Style::currentFont = "";
QString Style::currentJpFont = "";
int Style::currentFontSize = 12;
int Style::currentJpFontSize = 12;

Style* Style::_instance = new Style();



/*******************************************************
 *                    PUBLIC
 *******************************************************/

Styleinfo::Styleinfo()
{

}


Style::Style()
{

}

void Style::emitSignal(QString styleID)
{
    emit styleChanged(styleID);
}

void Style::signalFontChanged(QString font, QString jpfont, int size, int jpsize)
{
    emit fontChanged(font, jpfont, size, jpsize);
}

void Style::loadStyles()
{
    styles.clear();

    QDir styleDefaultDir (QString(dataFolder) + "styles/");
    QStringList styleFiles = styleDefaultDir.entryList(QStringList("*.ini"));

    foreach (QString styleFile, styleFiles){
        QString styleFileName = styleFile;
        styleFileName.chop(4);
        Styleinfo* styleinfo = new Styleinfo();

        QSettings styleSettings(QString(dataFolder) + "styles/" + styleFile, QSettings::IniFormat);
        styleSettings.setIniCodec("UTF-8");
        styleinfo->name = styleSettings.value("name").toString();
        styleinfo->author = styleSettings.value("author").toString();
        styleinfo->date = styleSettings.value("date").toString();
        styleinfo->desc = styleSettings.value("desc").toString();

        styles.insert(styleFileName, styleinfo);

    }

    setStyle();
}

/*!
 * \brief Style::getStyleInfo Get ID's and names of available style sheets.
 * \return A QHash of style-ID as key, and style-name as value. \n
 * Example: <DzStyle, DzCoding default style>, etc.
 */
QHash<QString, Styleinfo*> Style::getStyleInfo()
{
    return styles;
}


void Style::setConfigStyle(QString styleID)
{
    QSettings settings;//("DzCoding", "JapKatsuyou")
    settings.setValue("style", styleID);
}


QString Style::getConfigStyle()
{
    QSettings settings;//("DzCoding", "JapKatsuyou")

    return settings.value("style", "??").toString();
}


QString Style::getCurrentStyle()
{
    return currentstyle;
}


void Style::addReceiver(const QObject * receiver, const char * receiverSlot)
{
    connect(_instance, SIGNAL(styleChanged(QString)), receiver, receiverSlot);
}

void Style::addFontReceiver(const QObject *receiver, const char *receiverSlot)
{
    connect(_instance, SIGNAL(fontChanged(QString, QString, int,int)), receiver, receiverSlot);
}

QString Style::getConfigFont(bool jap)
{
    QSettings settings;

    if (jap)
        return settings.value("jp_font-family", "").toString();
    return settings.value("font-family", "").toString();
}

QString Style::getCurrentFont(bool jap)
{
    if (jap)
        return currentJpFont;

    return currentFont;
}

void Style::setConfigFont(QString font, bool jap)
{
    QSettings settings;

    if (jap)
        settings.setValue("jp_font-family", font);
    else
        settings.setValue("font-family", font);
}

int Style::getConfigFontSize(bool jap)
{
    QSettings settings;

    if (jap)
        return settings.value("jp_font-size", "12").toInt();

    return settings.value("font-size", "12").toInt();
}

int Style::getCurrentFontSize(bool jap)
{
    if (jap)
        return currentJpFontSize;

    return currentFontSize;
}

void Style::setConfigFontSize(int fontSize, bool jap)
{
    QSettings settings;

    if (jap)
        settings.setValue("jp_font-size", fontSize);
    else
        settings.setValue("font-size", fontSize);
}


/*******************************************************
 *                   PUBLIC SLOTS
 *******************************************************/

void Style::setStyle()
{
    //currentstyle = styleID;
    QString styleID = getConfigStyle();
    if(!styles.contains(styleID)){
        styleID = "DzStyle";
        setConfigStyle(styleID);
    }

    if (styleID != currentstyle){
        currentstyle = styleID;
        _instance->emitSignal(styleID);
        //qDebug() << "emit signal";
    }

}

void Style::setFont()
{
    QString font = getConfigFont(false);
    QString jpfont = getConfigFont(true);
    int size = getConfigFontSize(false);
    int jpsize = getConfigFontSize(true);

    bool changed = false;

    if (font != currentFont){
        currentFont = font;
        changed = true;
    }

    if (jpfont != currentJpFont){
        currentJpFont = jpfont;
        changed = true;
    }

    if (size != currentFontSize){
        currentFontSize = size;
        changed = true;
    }

    if (jpsize != currentJpFontSize){
        currentJpFontSize = jpsize;
        changed = true;
    }

    if (changed)
        _instance->signalFontChanged(font, jpfont, size, jpsize);
}
